src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -105,1002 +98,1361 @@
+                        } catch (OHServiceException e1) {
+                            OHServiceExceptionUtil.showMessages(e1);
+                        }
+                        if (inventory != null && !inventory.getStatus().equals(InventoryStatus.draft.toString())) {
+                            inventory.setStatus(InventoryStatus.draft.toString());
+                        }
+                        codeTextField.setText("");
+                    }
+                }
+            });
+        }
+        return codeTextField;
+    }
+
+    private void addInventoryRow(String code) throws OHServiceException {
+        List<MedicalInventoryRow> inventoryRowsList;
+        List<MedicalWard> medicalWardList = new ArrayList<>();
+        Medical medical;
+        if (wardId.isEmpty()) {
+            wardId = ((Ward) Objects.requireNonNull(wardComboBox.getSelectedItem())).getCode();
+        }
+        if (code != null) {
+            medical = medicalBrowsingManager.getMedicalByMedicalCode(code);
+            if (medical != null) {
+                medicalWardList = movWardBrowserManager.getMedicalsWard(wardId, medical.getCode(), false);
+            } else {
+                medical = chooseMedical(code);
+                if (medical != null) {
+                    boolean found = false;
+                    if (inventoryRowSearchList != null) {
+                        for (MedicalInventoryRow row : inventoryRowSearchList) {
+                            if (row.getMedical().getCode().equals(medical.getCode())) {
+                                found = true;
+                            }
+                        }
+                    }
+                    if (!found) {
+                        medicalWardList = movWardBrowserManager.getMedicalsWard(wardId, medical.getCode(), false);
+                    }
+                }
+            }
+        } else {
+            medicalWardList = movWardBrowserManager.getMedicalsWard(wardId, false);
+        }
+        inventoryRowsList = medicalWardList.stream().map(medWard -> new MedicalInventoryRow(0, medWard.getQty(),
+                medWard.getQty(), null, medWard.getMedical(), medWard.getLot())).toList();
+        if (inventoryRowSearchList == null) {
+            inventoryRowSearchList = new ArrayList<>();
+        }
+
+        inventoryRowsList.forEach(this::addMedInRowInInventorySearchList);
+        jTableInventoryRow.updateUI();
+    }
+
+    private Medical chooseMedical(String text) throws OHServiceException {
+        Map<String, Medical> medicalMap;
+        List<Medical> medicals = movWardBrowserManager.getMedicalsWard(wardId, false).stream().map(MedicalWard::getMedical).toList();
+
+        medicalMap = new HashMap<>();
+        for (Medical med : medicals) {
+            String key;
+            key = med.getCode().toString().toLowerCase();
+            medicalMap.put(key, med);
+        }
+        ArrayList<Medical> medList = new ArrayList<>();
```

Comment:
Concrete classes should not be used in the definition.
```suggestion
        List<Medical> medList = new ArrayList<>();
```